client/db.go

```diff
@@ -215,6 +215,15 @@ type Store interface {
 	// GetAllIndexes returns all the indexes that currently exist within this [Store].
 	GetAllIndexes(context.Context) (map[CollectionName][]IndexDescription, error)
 
+	// CreateDocIndex creates an index for the given document.
+	CreateDocIndex(context.Context, Collection, *Document) error
```

Comment:
suggestion: It feels like hosting these functions on `client.Collection` would be nicer, given that they have to take a `Collection` as a param anyway.

Reply:
that was our initial approach (Fred also took part in this design), but with all these http and cli wrappers it turned out to be more complicated. That's why wee took this route

Reply:
Why was that more complicated?  The bulk of the logic for handling collection stuff already exists in http/cli

Reply:
moved